webpack/components/extensions/RegistrationCommands/fields/ActivationKeys.js

```diff
@@ -39,47 +52,95 @@ const ActivationKeys = ({
 
   // Validate field when hostgroup is changed (host group may have some keys)
   useEffect(() => {
-    handleInvalidField('Activation Keys', akHasValidValue(hostGroupId, pluginValues?.activationKeys, hostGroupActivationKeys));
+    handleInvalidField(
+      'Activation Keys',
+      akHasValidValue(
+        hostGroupId,
+        pluginValues?.activationKeys,
+        hostGroupActivationKeys,
+      ),
+    );
   }, [handleInvalidField, hostGroupId, hostGroupActivationKeys, pluginValues]);
 
+  useEffect(() => {
+    if (activationKeys?.length === 1) {
+      updatePluginValues([activationKeys[0].name]);
+    }
+    if (activationKeys?.length === 0) {
+      setHasInteraction(true);
+    }
+  }, [activationKeys?.length]);
```

Comment:
In general, setting state in effects should be avoided if possible.

Rather, setting state in event handlers is much better. For `hasInteraction`, I think we can achieve the same thing by adding an `onFocus` to the `<FormGroup>`:

```js
<FormGroup
      onFocus={() => setHasInteraction(true)}
```

And for the `pluginValues`, I would strongly prefer if that could be updated outside an effect as well, but it seems slot and fill makes that too difficult. So I'm ok with keeping that part.

Either way, this is really good reading: https://react.dev/learn/you-might-not-need-an-effect